pkg/client/client.go

```diff
@@ -205,3 +214,44 @@ var addProviderVersionToUserAgent = request.NamedHandler{
 	Name: "openshift.io/cluster-api-provider-aws",
 	Fn:   request.MakeAddToUserAgentHandler("openshift.io cluster-api-provider-aws", version.Version.String()),
 }
+
+func resolveEndpoints(awsConfig *aws.Config, ctrlRuntimeClient client.Client, region string) error {
+	infra := &configv1.Infrastructure{}
+	infraName := client.ObjectKey{Name: globalInfrastuctureName}
+
+	if err := ctrlRuntimeClient.Get(context.Background(), infraName, infra); err != nil {
+		return err
+	}
+
+	// Do nothing when custom endpoints are missing
+	if infra.Spec.PlatformSpec.AWS == nil {
+		return nil
+	}
+
+	customEndpointsMap := buildCustomEndpointsMap(infra.Spec.PlatformSpec.AWS.ServiceEndpoints)
```

Comment:
Use the status and not the spec for these values.

Reply:
cc @alexander-demichev can we please address this?

Reply:
fixed